module/zfs/bplist.c

```diff
@@ -75,3 +75,17 @@ bplist_iterate(bplist_t *bpl, bplist_itor_t *func, void *arg, dmu_tx_t *tx)
 	}
 	mutex_exit(&bpl->bpl_lock);
 }
+
+void
+bplist_clear(bplist_t *bpl)
+{
+	bplist_entry_t *bpe;
+
+	mutex_enter(&bpl->bpl_lock);
+	while ((bpe = list_head(&bpl->bpl_list))) {
+		bplist_iterate_last_removed = bpe;
```

Comment:
I think that can go. Looking at the code, I didn't see anything that *used* bplist_iterate_last_removed, other than to assign to it.

Reply:
There's a comment above this saying that it's used for debugging - 
```
 /*
   * To aid debugging, we keep the most recently removed entry.  This way if
   * we are in the callback, we can easily locate the entry.
   */
```
so I think I'll leave it in for now.